invenio_rdm_records/collections/service.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-RDM is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""Collections service."""
+
+from invenio_communities.proxies import current_communities
+
+from .api import Collection, CollectionTree
+
+
+class CollectionItem:
+    """Collection item."""
+
+    def __init__(self, collection: Collection):
+        """Instantiate a Collection object.
+
+        Optionally pass a community to cache its information in the collection's instance.
+        """
+        self._collection = collection
+
+    def to_dict(self):
+        """Serialize the collection to a dictionary and add links."""
+        res = {**self._collection.to_dict()}
+        res[self._collection.id]["links"] = self.links
+
+        return res
+
+    @property
+    def links(self):
+        """Return the links of the collection."""
+        self_html = None
+        search = None
+        tree_slug = self._collection.collection_tree.slug
+        if self._collection.community:
+            self_html = f"/communities/{self._collection.community.slug}/{tree_slug}/{self._collection.slug}"
+            search = f"/api/communities/{self._collection.community.slug}/records"
+        else:
+            self_html = f"/collections/{tree_slug}/{self._collection.slug}"
+            search = "/api/records"
+        return {
+            "search": search,
+            "self_html": self_html,
+        }
+
+    @property
+    def community(self):
+        """Get the collection community."""
+        return self._collection.community
+
+    @property
+    def query(self):
+        """Get the collection query."""
+        return self._collection.query
+
+    @property
+    def title(self):
+        """Get the collection title."""
+        return self._collection.title
+
+
+class CollectionsService:
+    """Collections service."""
+
+    collection_cls = Collection
+
+    def create(self, identity, community_id, tree_slug, slug, title, query, **kwargs):
+        """Create a new collection."""
+        current_communities.service.require_permission(
```

Comment:
Maybe we should introduce a collection specific permissions instead of using the communities' ones?

`can_update` default value is `CommunityOwners`, is this correct?

Reply:
You are right, conceptually, communities are not required to have a collection (think about a collection on the repository level). However, the current state of this service is to require a community (this is our use case right now). 

I agree that the collections' permission policy at some point has to be specific for collections, but it still needs more thought and the service to be adapted to it. 

For a first launch, the permission `can_update` from communities is good since we don't want any user of a community creating collections for a community